intercom_flutter_platform_interface/lib/intercom_status_callback.dart

```diff
@@ -0,0 +1,33 @@
+class IntercomStatusCallback {
+  /// callback when intercom operation is failed.
+  /// will contain the error information.
+  final Function(IntercomError error)? onFailure;
+
+  /// callback when intercom operation is success.
+  final Function()? onSuccess;
+
+  /// class for intercom status to check if the operation is success or failure.
+  /// if the operation failed then [onFailure] callback will be executed with
+  /// [IntercomError] details.
+  ///
+  IntercomStatusCallback({
+    this.onSuccess,
+    this.onFailure,
+  });
+}
+
+class IntercomError {
+  /// error code
+  final int errorCode;
+
+  /// error message
+  final String errorMessage;
+
+  /// class for the Intercom error data.
+  IntercomError(this.errorCode, this.errorMessage);
+
+  @override
+  String toString() {
```

Comment:
Is it required to override `toString` here?

Reply:
Yes, it is required to check or print the error data.